uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChange.php

```diff
@@ -59,15 +62,15 @@ public static function getBalanceForEntitlement($entitlementID, $leaveRequestSta
       FROM {$balanceChangeTable} leave_balance_change
       LEFT JOIN {$leaveRequestDateTable} leave_request_date 
              ON leave_balance_change.source_id = leave_request_date.id AND 
-                leave_balance_change.source_type = 'leave_request_day'
+                leave_balance_change.source_type = '". self::SOURCE_LEAVE_REQUEST_DAY ."'
```

Comment:
it is better to stick with curly braces style since you used it in the rest of the code {self::SOURCE_LEAVE_REQUEST_DAY} or maybe it it doesn't work with non variables ?